api/flaskr/__init__.py

```diff
@@ -111,16 +119,52 @@ def login():
         if login_user:
             if bcrypt.check_password_hash(login_user['password'], password):
                 session['username'] = username
-                session['access_token'] = create_access_token(identity=username)
-                session['refresh_token'] = create_access_token(identity=username)
-                
-                return redirect(url_for('index'))
+                expires = datetime.timedelta(hours=1)
+                tokens = { 
+                    'access_token': create_access_token(identity=username, expires_delta=expires),
+                    'refresh_token': create_refresh_token(identity=username) 
+                }
+                return jsonify(tokens), 200
 
         return Response('Username and password do not match', status=401)
 
-    @app.route('/getSession', methods=(['GET']))
+    @app.route('/refresh', methods=['POST'])
+    @jwt_refresh_token_required
+    def refresh():
+        current_user = get_jwt_identity()
+        ret = {
+            'access_token': create_access_token(identity=current_user)
+        }
+        return jsonify(ret), 200
+
+    # Endpoint for revoking the current users access token
+    @app.route('/logout', methods=['DELETE'])
+    @jwt_required
+    def logout():
+        jti = get_raw_jwt()['jti']
+        blacklist.add(jti)
+        return jsonify({"msg": "Successfully logged out"}), 200
+
+
+    # Endpoint for revoking the current users refresh token
+    @app.route('/logout2', methods=['DELETE'])
+    @jwt_refresh_token_required
+    def logout2():
+        jti = get_raw_jwt()['jti']
+        blacklist.add(jti)
+        return jsonify({"msg": "Successfully logged out"}), 200
```

Comment:
There is duplicate code for refresh token. Isn't there a way we can handle both refresh token and access token in a single endpoint?

Reply:
The code here was taken directly from the flask-jwt-extended docs. Upon research I found a similar [issue](https://github.com/vimalloc/flask-jwt-extended/issues/53). Maybe we can change the logout logic as suggested [here](https://github.com/vimalloc/flask-jwt-extended/issues/53#issuecomment-308612135). Currently, to logout we would need to make 2 requests from frontend.